tensorflow_probability/python/distributions/sratio_test.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2020 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import itertools
+# Dependency imports
+from absl.testing import parameterized
+import numpy as np
+import tensorflow.compat.v2 as tf
+import tensorflow_probability as tfp
+
+from tensorflow_probability.python.internal import test_util
+
+tfd = tfp.distributions
+tfb = tfp.bijectors
+
+
+@test_util.test_all_tf_execution_regimes
+class StoppingRatioLogisticTest(test_util.TestCase):
+
+  def _random_cutpoints(self, shape):
+    return self._ordered.inverse(self._rng.randn(*shape))
+
+  def _random_location(self, shape):
+    return self._rng.randn(*shape)
+
+  def setUp(self):
+    self._ordered = tfb.Ordered()
+    self._rng = np.random.RandomState(test_util.test_seed())
+    super(StoppingRatioLogisticTest, self).setUp()
+
+  @parameterized.parameters(
+      itertools.product(['cutpoints', 'loc', 'both'], [[], [1], [1, 2, 3]])
+  )
+  def testBatchShapes(self, test, batch_shape):
+    if test == 'cutpoints':
+      cutpoints = self._random_cutpoints(batch_shape + [2])
+      loc = tf.constant(0., dtype=tf.float64)
+    elif test == 'loc':
+      cutpoints = tf.constant([1., 2.], dtype=tf.float64)
+      loc = self._random_location(batch_shape)
+    elif test == 'both':
+      cutpoints = self._random_cutpoints(batch_shape + [2])
+      loc = self._random_location(batch_shape)
+
+    dist = tfd.StoppingRatioLogistic(cutpoints=cutpoints, loc=loc)
+
+    self.assertAllEqual(dist.batch_shape, batch_shape)
+    self.assertAllEqual(
+        self.evaluate(dist.batch_shape_tensor()), batch_shape)
+
+    self.assertAllEqual(dist.event_shape, [])
+    self.assertAllEqual(self.evaluate(dist.event_shape_tensor()), [])
+
+    categorical_probs = dist.categorical_probs()
+    categorical_probs_shape = tf.shape(categorical_probs)
+    self.assertAllEqual(
+        self.evaluate(categorical_probs_shape), batch_shape + [3])
+
+    sample_shape = tf.shape(dist.sample(seed=test_util.test_seed()))
+    self.assertAllEqual(self.evaluate(sample_shape), batch_shape)
+
+    sample_shape_n = tf.shape(
+        dist.sample([4, 5], seed=test_util.test_seed()))
+    self.assertAllEqual(self.evaluate(sample_shape_n), [4, 5] + batch_shape)
+
+  def testProbs(self):
```

Comment:
Can you add batch shapes to these tests? This will let us test that the vectorization is working properly.

Reply:
Just to be sure, I am testing batch shapes for all important functions above. Should I still dot it for `testProbs` and the other functions?